staff/dario-fernandez/pre-course/pasapalabra-html/index.js

```diff
@@ -0,0 +1,243 @@
+            }      
+        }
+    },
+    pasapalabra() {
+        JUEGO.cajaTexto.value = 'pasapalabra'
+        JUEGO.turnoLetra()
+    },
+    temporizador() {
+        if(JUEGO.letrasRestantes) {
+            if(JUEGO.segundosRestantes >= 0) {
+                    JUEGO.temporizadorHTML.innerHTML = JUEGO.segundosRestantes
+                    JUEGO.segundosRestantes--
+                    setTimeout(JUEGO.temporizador, 1000)       
+            } else {
+                JUEGO.finJuego()
+            }
+        }
+    },
+    nuevoJuego() {
+        this.mostrarJuego()
+        this.pintarPregunta()
+        this.eventosOn()
+        this.cajaTexto.focus()
+        this.temporizador()
+    },
+    finJuego(){
+        JUEGO.ocultarJuego()
+        JUEGO.eventosOff()
+        this.cajaTexto.value = ''
+        POSTJUEGO.nuevoPostjuego()
+    }
+}
+
+
+
+var POSTJUEGO = {
+    rankingHTML: document.getElementById('ranking'),
+    ranking: [],
+    record: [],
+    pantalla: document.getElementById('postjuego'),
+    puntuacionHTML: document.getElementById('puntuacion'),
+    botonNuevaPartida: document.getElementById('nueva-partida'),
+    mostrarPostjuego() {
+        this.pantalla.classList.remove('postjuego-container--oculto')
+        this.pantalla.classList.add('postjuego-container--visible')
+    },
+    ocultarPostjuego() {
+        this.pantalla.classList.remove('postjuego-container--visible')
+        this.pantalla.classList.add('postjuego-container--oculto')
+    },
+    eventosOn() {
+        this.botonNuevaPartida.addEventListener('click', this.finPostjuego)
+    },
+    eventosOff() {
+        this.botonNuevaPartida.removeEventListener('click', this.finPostjuego)
+    },
+    resetRosco() {
+        for(let i = 0; i < ROSCO.length; i++) {
+            ROSCO[i].status = 0
+        }
+    },
+    resetRoscoHTML() {
+        for(let i = 0; i < ROSCO.length; i++) {
+            document.getElementById(ROSCO[i].letter).classList.remove('rosco__letra--acertada')
+            document.getElementById(ROSCO[i].letter).classList.remove('rosco__letra--fallada')
+            document.getElementById(ROSCO[i].letter).classList.remove('rosco__letra--focus')
+        }
+    },
+    resetVariables() {
+        JUEGO.letrasRestantes = ROSCO.length
+        JUEGO.segundosRestantes = 100
+        JUEGO.aciertos = 0
+        JUEGO.fallos = 0
+        JUEGO.i = 0
+    },
+    mostrarPuntuacion() {
+        this.puntuacionHTML.innerHTML = `${JUEGO.aciertos} aciertos<br>${JUEGO.fallos} fallos<br>${100 - JUEGO.segundosRestantes} segundos`
+    },
+    resetJuego() {
+        this.resetRosco()
+        this.resetRoscoHTML()
+        this.resetVariables()
+    },
+    nuevoPostjuego() {
+        this.mostrarPostjuego()
+        this.eventosOn()
+        this.mostrarPuntuacion()
+    },
+    finPostjuego() {
+        POSTJUEGO.resetJuego()
+        POSTJUEGO.eventosOff()
+        POSTJUEGO.ocultarPostjuego()
+        PREJUEGO.nuevoPrejuego()
+    },
+}
```

Comment:
Creo que la única pega que encuentro (y ni siquiera debe de afectar al código) es la falta de ; por protocolos de precaución. Por lo demás creo que algún día de estos te preguntaré sobre el "this" porque claramente lo dominas y es algo que cojeo, good fucking job